net/tcp/tcp_seqno.c

```diff
@@ -142,6 +143,31 @@ uint32_t tcp_addsequence(FAR uint8_t *seqno, uint16_t len)
 
 void tcp_initsequence(FAR uint8_t *seqno)
 {
+  int ret;
+
+  /* Get a random TCP sequence number */
+
+  ret = getrandom(&g_tcpsequence, sizeof(uint32_t) - 1, 0);
```

Comment:
Why -1 here?

Reply:
Because I'm ++g_tcpsequence to avoid g_tcpsequence == 0

Reply:
But current behavior will be different for BE and LE archs. You request to fill 3 bytes of uint32_t value with random data, but it will be 3 bytes at the beginning or at the end depending if the system is big or little endian.

Reply:
I need to spend more time reviewing this

Reply:
It was a mistake, I was thinking it as UINT_MAX not as sizeof(uint32_t). Fixed